otaku_back/otaku_back/database/repository.py

```diff
@@ -0,0 +1,34 @@
+from typing import List, Optional
+from django.db import models
+
+class Repository[T: models.Model]:
+    def __init__(self, model: type[T]):
+        self._model = model
+
+    def get_all(self) -> List[T]:
+        return self._model.objects.all()
+
+    def get_by_id(self, id: int) -> Optional[T]:
```

Comment:
remove id int type in order to support ObjectId, string and id types (leave untyped)